chipsec/chipset.py

```diff
@@ -782,18 +783,26 @@ def is_register_defined(self, reg_name):
 
     def get_register_def(self, reg_name, bus_index=0):
         reg_def = self.Cfg.REGISTERS[reg_name]
-        if (reg_def["type"] == "pcicfg" or reg_def["type"] == "mmcfg") and ("device" in reg_def):
+        if "device" in reg_def:
             dev_name = reg_def["device"]
-            if dev_name in self.Cfg.CONFIG_PCI:
-                dev = self.Cfg.CONFIG_PCI[dev_name]
-                reg_def['bus'] = dev['bus']
-                reg_def['dev'] = dev['dev']
-                reg_def['fun'] = dev['fun']
-                if dev_name in self.Cfg.BUS:
-                    if bus_index < len(self.Cfg.BUS[dev_name]):
-                        reg_def['bus'] = self.Cfg.BUS[dev_name][bus_index]
-                    else:
-                        logger().error( "Bus index {:d} for '{}' not found.".format(bus_index, dev_name) )
+            if reg_def["type"] == "pcicfg" or reg_def["type"] == "mmcfg":
+                if dev_name in self.Cfg.CONFIG_PCI:
+                    dev = self.Cfg.CONFIG_PCI[dev_name]
+                    reg_def['bus'] = dev['bus']
+                    reg_def['dev'] = dev['dev']
+                    reg_def['fun'] = dev['fun']
+                    if dev_name in self.Cfg.BUS:
+                        if bus_index < len(self.Cfg.BUS[dev_name]):
+                            reg_def['bus'] = self.Cfg.BUS[dev_name][bus_index]
+                        else:
+                            logger().error( "Bus index {:d} for '{}' not found.".format(bus_index, dev_name) )
+            elif reg_def["type"] == "memory":
```

Comment:
Does `<memory>` have a `device` attribute?  Was this for `<mmio>`?

Reply:
this is for defining the register with a device type memory.  Using the common.xml for example:
now I can create a register based upon TPM as follows:
<register name ="TEST" type="memory" device="TPM" offset="0x0" size="4" desc="Test" />